src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/PropertyGridViewRowsAccessibleObjectTests.cs

```
@@ -10,81 +10,96 @@ namespace System.Windows.Forms.Tests.AccessibleObjects
 {
     public class PropertyGridViewRowsAccessibleObjectTests : IClassFixture<ThreadExceptionFixture>
     {
-        [WinFormsFact(Skip = "Suspect causing deadlocks, see: https://github.com/dotnet/winforms/issues/3095")]
+        [WinFormsFact]
         public void PropertyGridViewRowsAccessibleObject_Ctor_Default()
         {
-            TestForm form = new TestForm();
-            Application.Run(form);
-
-            const int topBorder = 1;
             const int bottomBorder = 1;
-            int entriesBorders = form.entriesBorders;
-
-            Assert.True(form.AccRowHeightSum == form.AccPropertyGridViewHeight - topBorder - bottomBorder - entriesBorders);
-        }
-    }
+            const int topBorder = 1;
 
-    public class TestForm : Form
-    {
-        public int AccPropertyGridViewHeight { get; set; }
-        public int AccRowHeightSum { get; set; }
+            // track that the form has been shown and executed the main assert
+            bool formShown = false;
 
-        DomainUpDown domainUpDown;
-        PropertyGrid propertyGrid;
+            using TestForm form = new TestForm();
+            form.Load += (s, e) =>
+            {
+                Dimensions dimensions = form.CalculateDimensions();
+                Assert.True(dimensions.AccRowHeightSum == dimensions.AccPropertyGridViewHeight - topBorder - bottomBorder - dimensions.EntriesBorders);
```

Don't do an Assert in Form.Load, exceptions aren't guaranteed to propagate out of a Form.Load event. Test failures will not be recognized without observing an exception.

---

I fixed this but I'm leaving the comment unresolved for visibility, its important to know how unreliable Form.Load is - its executed while holding a lock so prone to deadlock if you have multiple UI threads, and exceptions are often (always?) swallowed so you shouldn't be doing anything that may be throwing, in particular no test asserts.